### DISCUSSION TITLE
What would you need for your secrets?

### DISCUSSION THREAD

**Discussion by dolanor** [dagger/dagger#4910] [Help]
We have more improvements planned to the secrets API, as presented in the 0.5 release.
- [improved scrubbing strategy](https://github.com/dagger/dagger/issues/4864)
- [more control over secret mounts](https://github.com/dagger/dagger/issues/3323)
- [pass secrets to Dockerfile build](https://github.com/dagger/dagger/issues/4289)

Maybe, there is something we didn't think about.

Please vote, share your opinions and ideas so we can prioritize accordingly.

**Comment by dolanor** [comment]
[improved scrubbing strategy](https://github.com/dagger/dagger/issues/4864)

**Comment by dolanor** [comment]
[more control over secret mounts](https://github.com/dagger/dagger/issues/3323)

**Comment by dolanor** [comment]
[pass secrets to Dockerfile build](https://github.com/dagger/dagger/issues/4289)

**Comment by skycaptain** [comment]
It would be great if we could limit secrets to single `with_exec` commands, like [`RUN --mount=type=secret`](https://docs.docker.com/engine/reference/builder/#run---mounttypesecret), e.g.: 

```python
my_secret = client.set_secret("PIP_INDEX_URL", "my-secret-url")
ctr = (
    client.container().from_("python:3")
    .with_exec(["pip3", "install", "-r", "requirements.txt"], secrets=[my_secret])
    ...
)
```
`my-secret` would only be available for the pip install command but not for subsequent commands.
